File: src/main/java/com/aws/greengrass/localdebugconsole/messageutils/CommunicationMessage.java
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

package com.aws.greengrass.localdebugconsole.messageutils;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;

/**
 * Class for modeling communication messages. PubSub or MQTT message.
 */
@EqualsAndHashCode
@AllArgsConstructor
@Getter
public class CommunicationMessage implements Comparable<CommunicationMessage> {
    private final String subId;
    private final String subscribedTopic;
    private final String topic;
    private final String payload;

    // for unit testing
    @Override
    public int compareTo(CommunicationMessage o) {
        // Only both of them equals return 1 * 1
        return topic.compareTo(o.topic) * payload.compareTo(o.payload);
    }
}

File: src/main/js/dashboard-frontend/src/util/CommunicationMessage.ts
/*
 * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
 * SPDX-License-Identifier: Apache-2.0
 */

export class CommunicationMessage {
  subId: string;
  subscribedTopic: string;
  topic: string;
  payload: string;
  constructor(
    subId: string,
    subscribedTopic: string,
    topic: string,
    payload: string,
  ) {
    this.subId = subId;
    this.subscribedTopic = subscribedTopic;
    this.topic = topic;
    this.payload = payload;
  }
}
